src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1767,11 +1865,15 @@ bool entitiesQuery
       // FIXME P4: Once issue #1705 is implemented, this check can be removed.
       if (qStringFilters(sco->value, filters) != true)
       {
-        if (badInputP)
+        if (badInputP)  // Bad Input reported by higher level
         {
           *badInputP = true;
           *err       = "invalid query expression";
         }
+        else
```

Comment:
Why this else brach has been added? I mean, it don't see the relation with the atof() -> str2double() refactoring...

Maybe is due to some .test get broken?